pkgs/development/guile-modules/guile-json/default.nix

```diff
@@ -15,19 +15,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-xTSaI4D2fIphOps698mNITJdRDAjNp5vdhs2bpaUaEM=";
   };
 
-  postConfigure = ''
-    sed -i '/moddir\s*=/s%=.*%=''${out}/share/guile/site%' Makefile;
-    sed -i '/objdir\s*=/s%=.*%=''${out}/share/guile/ccache%' Makefile;
-    sed -i '/moddir\s*=/s%=.*%=''${out}/share/guile/site/json%' json/Makefile;
-    sed -i '/objdir\s*=/s%=.*%=''${out}/share/guile/ccache/json%' json/Makefile;
-  '';
-
   nativeBuildInputs = [
     pkg-config texinfo
   ];
   buildInputs = [
     guile
   ];
+  doCheck = true;
+  makeFlags = [ "GUILE_AUTO_COMPILE=0" ];
```

Comment:
I couldn't find why we want to set this or what this does.

Reply:
This is addressed in the review with similar changes at #189421 ([comment link](https://github.com/NixOS/nixpkgs/pull/189421#discussion_r962424917)).